Cc/activate cta

camdencheek opened the pull request:
<!-- PR description tips: https://www.notion.so/sourcegraph/Write-a-good-pull-request-description-610a7fd3e613496eb76f450db5a49b6e -->

## Test plan

<!-- REQUIRED; info at https://docs-legacy.sourcegraph.com/dev/background-information/testing_principles -->

## Changelog

<!-- OPTIONAL; info at https://www.notion.so/sourcegraph/Writing-a-changelog-entry-dd997f411d524caabf0d8d38a24a878c -->

taiyab reviewed (commented):
First off: incredible work @camdencheek, this is looking slick!

On review of everything, I think we can afford to, right now, actually show the intro modal every time someone reactivates the new experience. Because we're not expecting people to constantly move back and forth, the extra button click is a decent tradeoff for both the additional instruction/context and sense of confirmation the modal gives you.

A few little niggles:
- we're missing the CMD+K badge next to the "reworked fuzzy finder" title (that's my bad, I didn't make sure it was in both light/dark designs)
- we're missing a way to prep users to think about leaving us feedback on that modal state (although I'm assuming this could just be something you decided to skip for this minimal version, in which case, we can add a line of copy to the small subtext below the "Awesome, I'm ready to use it!" button telling users they can also leave feedback
- the modal backgrounds don't look like they match up with the designs
- we could consider linking to the discord, although I suspect that's more for the PLG audience and would look pretty scrappy for enterprise folks (can you confirm this @jdorfman?)

Not directly related to this PR: would it be possible to rehouse the "developer settings" UI into the profile dropdown too, to ensure we don't have that taking up valuable header space, especially when we want the default React version to also use the minimal navigation? There also might be other states/things like that that clog up the header area that we might want to accommodate for.

Overall, incredible stuff. All my feedback is non-blocking to merge this.

jdorfman commented:
@taiyab 

> we could consider linking to the discord, although I suspect that's more for the PLG audience and would look pretty scrappy for enterprise folks (can you confirm this @jdorfman?)

Let's use the forum: https://community.sourcegraph.com/c/code-search/9

camdencheek commented:
> I think we can afford to, right now, actually show the intro modal every time someone reactivates the new experience

Sounds good to me. I assume we do not want to show it to users for which it's on by default?

> we're missing the CMD+K badge next to the "reworked fuzzy finder" title (that's my bad, I didn't make sure it was in both light/dark designs)

Oops, my bad. I added that after taking the video (it's in the screenshots). 

> we're missing a way to prep users to think about leaving us feedback on that modal state

Oops, yep. Forgot that in the "followups" section. Implementation is surprisingly tricky on that one

> the modal backgrounds don't look like they match up with the designs

Whoops, will adjust before merge. 

> rehouse the "developer settings" UI

That only exists in local dev. Customers should never see that. Note I removed it when I took the screenshots because it's distracting 🙂

taiyab commented:
> > I think we can afford to, right now, actually show the intro modal every time someone reactivates the new experience
> 
> Sounds good to me. I assume we do not want to show it to users for which it's on by default?

We can, but just the first time. So everyone gets to see it once at least.
 
> > rehouse the "developer settings" UI
> 
> That only exists in local dev. Customers should never see that. Note I removed it when I took the screenshots because it's distracting 🙂

Sweet!

bahrmichael commented on client/web/src/sveltekit/SvelteKitNavItem.tsx:
```diff
@@ -13,28 +15,92 @@ import styles from './SvelteKitNavItem.module.scss'
+                                </Button>
+                            </H3>
+                            <Text>
+                                Remember, you can always switch back using the toggle above. It’ll keep getting improved
```
(non-blocking)

What do you think about changing the wording from "It’ll keep getting improved" to "We continue to improve it"?

camdencheek replied:
updated!

bahrmichael commented on client/web/src/sveltekit/SvelteKitNavItem.tsx:
```diff
@@ -13,28 +15,92 @@ import styles from './SvelteKitNavItem.module.scss'
+                        <H3>What's this "New, faster UX"?</H3>
+                        <Text>
+                            We've been busy at work on a new Code Search experience, built from the ground up for
+                            performance, which now available in beta.
```
"which IS now available in beta"

bahrmichael reviewed (changes requested):
Code looks good to me. I'd like to see the "which IS now available in beta" fixed, but apart from that good to go.

thenamankumar commented on client/shared/src/settings/temporary/TemporarySettings.ts:
```diff
@@ -96,6 +96,11 @@ export interface TemporarySettingsSchema {
+
+    'webNext.welcomeOverlay.dismissed': boolean
+    'webNext.departureMessage.dismissed': boolean
+    'webNext.toggled.on': boolean // whether the user has explicitly enabled with the toggle
```
Can we not turn `webNext.toggled.on` and `webNext.toggled.on`  into a single setting? By default, `undefined` means not set.

camdencheek replied:
Hmm, you might be right. I think this behavior is going to change a bit based on Taiyab's feedback, but I'll take a look at consolidating

rrhyne commented:
There is some valuable info here. What if below the description we offer a quick link to open the intro again?

taiyab commented:
> There is some valuable info here. What if below the description we offer a quick link to open the intro again?

Semi relevant to this: we're gonna show this every time you switch into the new experience (rather than once). Seeing as people are unlikely to be constantly switching back and forth, I thought the tradeoff was worth hitting an extra button each time to get that import primer info upfront each time.

camdencheek commented:
> show the modal every time you switch into the new experience

PR is updated with this behavior now btw

peterguy reviewed (approved):
Lots of comments and suggestions; nothing blocking. Approving to unblock merging.
